scripts/validate_rst_title_capitalization.py

```diff
@@ -184,12 +189,20 @@ def correct_title_capitalization(title: str) -> str:
     # first word character.
     correct_title: str = re.sub(r"^\W*", "", title).capitalize()
```

Comment:
I know you didn't introduce this (maybe it was me), but don't you think `correct_title` is misleading here? I'd probably overwrite title instead. Also, it may help read this function is in these steps we have a comment with an example of what each step is doing.

I guess removing this line temporary should make the script fail with the examples that this is making pass. I personally think it'd be helpful if for the non trivial steps we had comments with what it does.

Like `- foo bar foo-bar`, then `foo bar foo-bar`, then `['foo', 'bar', 'foo-bar']`...